.github/scripts/arm/build.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+
+[ -z "$1" ] && { echo "Missing 1st argument: PostgREST github commit SHA"; exit 1; }
+[ -z "$2" ] && { echo "Missing 2nd argument: Docker repo"; exit 1; }
+[ -z "$3" ] && { echo "Missing 3rd argument: Docker username"; exit 1; }
+[ -z "$4" ] && { echo "Missing 4th argument: Docker password"; exit 1; }
+[ -z "$5" ] && { echo "Missing 5th argument: Build environment directory name"; exit 1; }
+
+PGRST_GITHUB_COMMIT="$1"
+DOCKER_REPO="$2"
+DOCKER_USER="$3"
+DOCKER_PASS="$4"
+DOCKER_BUILD_PATH="$5"
+
+clean_env()
+{
+    sudo docker logout
+}
+
+# Login to Docker
+sudo docker logout
+{ echo $DOCKER_PASS | sudo docker login -u $DOCKER_USER --password-stdin; } || { echo "Couldn't login to docker"; exit 1; }
+
+trap clean_env sigint sigterm exit
+
+# Move to the docker build environment
+cd ~/$DOCKER_BUILD_PATH
+
+# Build ARM versions
+sudo docker buildx build --build-arg PGRST_GITHUB_COMMIT=$PGRST_GITHUB_COMMIT \
+                         --build-arg BUILDKIT_INLINE_CACHE=1 \
+                         --platform linux/arm/v7,linux/arm64 \
+                         --cache-from $DOCKER_REPO/postgrest:postgrest-build-arm \
+                         --target=postgrest-build \
+                         -t $DOCKER_REPO/postgrest:postgrest-build-arm \
+                         --push .
```

Comment:
We also have a `docker login` and `push` on:

https://github.com/PostgREST/postgrest/blob/cb145271ac8136354df0800e689601c5debd7bfd/.github/workflows/ci.yaml#L366-L370

Would be nice to DRY this up into a nix script.

Not sure if it can be done for `push` since you use it inline with `buildx` here(haven't used that before) but seems doable for `login`.

Reply:
In this case, the docker login in `build.sh` is used to publish the intermediate ARM build image for caching purposes. The login in `docker-publish.sh` is the one that is duplicated as you mentioned. Applying a DRY here would solve the need to manually delete the final ARM PostgREST image, but I can't see how to make it work with nix and multiple architectures: the `amd64` version only needs a scratch image while the `arm` ones need a bigger ecosystem. 

Maybe it should be left for later or someone can find a better solution here.